### DISCUSSION TITLE
[OpenSSL] Integrating `pyopenssl` for `proxy.common.pki` instead of using `openssl` command line.

### DISCUSSION THREAD

**Discussion by hash3liZer** [abhinavsingh/proxy.py#805] [Ideas]
I ran into this issue while doing TLS interception on windows. If you don't have openssl installed, you will run into errors while decrypting traffic on the fly. During interception, the functions from `proxy.common.pki` are heavily used (probably to create and manage private & public keys). 

My end goal is to distribute a compiled binary package of a project using `proxy.py`. However, now i have to make sure that openssl is bundled with it. And it is eating too much space for me. 

So, upon reading the code from `proxy.common.pki`, the functions within are using command line version openssl. While python does provide `pyopenssl` library which can be used with cross platform support. '

So, my feature request would be to use `pyopenssl` instead of command line version. So, we don't have any kind of dependencies while using `proxy.py`

**Comment by abhinavsingh** [comment]
@hash3liZer Yes this is a possible direction for future.  What we can do is build an interface out of existing `proxy.common.pki` functionality.  Then write implementations using `pyopenssl`, `openssl` and/or other SSL libraries.  Then configure proxy.py to use one of these implementations at runtime.

Let's keep this open and tackle in near future.  Let me know if you make any progress on this front.
